src/dep.mli

```diff
@@ -4,19 +4,30 @@ type t
 
 val file : Path.t -> t
 val env : Env.Var.t -> t
+val universe : t
 
 val compare : t -> t -> Ordering.t
 
 val pp : t Fmt.t
 
+module Trace : sig
+  type 'a t =
+    | Changed of 'a
```

Comment:
I'm not sure the concept of `Changed` belongs to this module. What about simply doing a `Dep.Set.has_universe deps` test in `build_system.ml`?

Reply:
Yeah, I was hoping that there would be a to factor out all this trace comparison and marshal stuff into its own module, but my attempt here isn't cutting it anyway. I'll keep it simple.